File: Assignment35_1.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : CountCapital
// Description   : Count the number of capital characters (A-Z) from string
// Input         : String
// Output        : Integer (Count of capital letters)
// Author        : Raj Samir Jadhav
// Date          : 28/11/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public int CountCapital(String str)          // Function
    {
        int iCnt = 0;                           // Counter

        for(int i = 0; i < str.length(); i++)   // Loop till end of string
        {
            char ch = str.charAt(i);

            if(ch >= 'A' && ch <= 'Z')          // Check if character is Capital
            {
                iCnt++;
            }
        }
        return iCnt;                             // Return count
    }
}

class Assignment34_1
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        int iRet = pobj.CountCapital(input);

        System.out.println("Number of capital characters : " + iRet);  // Display Result

        sobj.close();
    }
}

File: Assignment35_1.c
#include <stdio.h>
///////////////////////////////////////////////////////////////////////////////
// Function Name : Display
// Description   : Count Capital letter in string
// Input         : String
// Output        : Count of Capital letters
// Author        : Raj Samir Jadhav
// Date          : 3/12/2025
///////////////////////////////////////////////////////////////////////////////


int CountCapital(char *str)                      // Function to count capital letters in string
{
    int iCnt = 0;
    while(*str != '\0')                          // Traverse string till null character
{
        if((*str >= 'A') && (*str <= 'Z'))        // Check if character is Capital (A-Z)
        {
            iCnt++;
        }
        str++;                                    // Move pointer to next character
    }

    return iCnt;                                 // Return count
}

int main()
{
    char arr[20];
    int iRet = 0;

    printf("Enter string : ");
    scanf("%[^'\n']s",arr);                      // Accept string till newline

    iRet = CountCapital(arr);

    printf("Number of capital characters : %d\n",iRet);

    return 0;
}
